lib/internal/messages.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+// adapted from the approach documented at:
+// http://jaysoo.ca/2014/03/20/i18n-with-es6-template-strings/
```

Comment:
Should we maybe note why we are doing it like this a bit more here?

Also since this won;t be in the API docs it may be useful to have a quick API-doc thing here. (i.e. what to put in, what comes out)